src/base.jl

```diff
@@ -35,18 +35,19 @@ Base.isvalid(s::ShortString, i::Integer) = isvalid(String(s), i)
 Base.iterate(s::ShortString) = iterate(String(s))
 Base.iterate(s::ShortString, i::Integer) = iterate(String(s), i)
 Base.lastindex(s::ShortString) = sizeof(s)
-Base.ncodeunits(s::ShortString) = ncodeunits(String(s))
+Base.ncodeunits(s::ShortString) = sizeof(s)
 Base.print(s::ShortString) = print(String(s))
 Base.show(io::IO, str::ShortString) = show(io, String(str))
-Base.sizeof(s::ShortString{T}) where T = Int(s.size_content & size_mask(T))
+Base.sizeof(s::ShortString{T}) where T = Int(s.size_content & (size_mask(s) % UInt))
```

Comment:
Please help me understand why is `%` necessary? I must be missing something obvious.

Reply:
I got it.